scala_proto/scala_proto.bzl

```diff
@@ -0,0 +1,442 @@
+    )
+
+    native.bind(
+        name = 'io_bazel_rules_scala/dependency/proto/netty_buffer',
+        actual = '@scala_proto_rules_netty_buffer//jar'
+    )
+
+    native.maven_jar(
+        name = "scala_proto_rules_netty_transport",
+        artifact = "io.netty:netty-transport:4.1.8.Final",
+        sha1 = "905b5dadce881c9824b3039c0df36dabbb7b6a07",
+        server = "scala_proto_deps_maven_server",
+    )
+
+    native.bind(
+        name = 'io_bazel_rules_scala/dependency/proto/netty_transport',
+        actual = '@scala_proto_rules_netty_transport//jar'
+    )
+
+    native.maven_jar(
+        name = "scala_proto_rules_netty_resolver",
+        artifact = "io.netty:netty-resolver:4.1.8.Final",
+        sha1 = "2e116cdd5edc01b2305072b1dbbd17c0595dbfef",
+        server = "scala_proto_deps_maven_server",
+    )
+
+    native.bind(
+        name = 'io_bazel_rules_scala/dependency/proto/netty_resolver',
+        actual = '@scala_proto_rules_netty_resolver//jar'
+    )
+
+    native.maven_jar(
+        name = "scala_proto_rules_netty_common",
+        artifact = "io.netty:netty-common:4.1.8.Final",
+        sha1 = "ee62c80318413d2375d145e51e48d7d35c901324",
+        server = "scala_proto_deps_maven_server",
+    )
+
+    native.bind(
+        name = 'io_bazel_rules_scala/dependency/proto/netty_common',
+        actual = '@scala_proto_rules_netty_common//jar'
+    )
+
+    native.maven_jar(
+        name = "scala_proto_rules_netty_handler_proxy",
+        artifact = "io.netty:netty-handler-proxy:4.1.8.Final",
+        sha1 = "c4d22e8b9071a0ea8d75766d869496c32648a758",
+        server = "scala_proto_deps_maven_server",
+    )
+
+    native.bind(
+        name = 'io_bazel_rules_scala/dependency/proto/netty_handler_proxy',
+        actual = '@scala_proto_rules_netty_handler_proxy//jar'
+    )
+
+def _colon_paths(data):
+  return ':'.join([f.path for f in data])
+
+def _gen_proto_srcjar_impl(ctx):
+    acc_imports = depset()
+    for target in ctx.attr.deps:
+        acc_imports += target.proto.transitive_sources
+
+    # Command line args to worker cannot be empty so using padding
+    flags = ["-"]
+    if ctx.attr.with_grpc:
+        flags.append("grpc")
+    if ctx.attr.with_java_conversions:
+        flags.append("java_conversions")
+    if ctx.attr.with_flat_package:
+        flags.append("flat_package")
+    if ctx.attr.with_single_line_to_string:
+        flags.append("single_line_to_string")
+
+    worker_content = "{output}\n{paths}\n{flags_arg}".format(
+        output = ctx.outputs.srcjar.path,
+        paths = _colon_paths(acc_imports),
+        flags_arg = ",".join(flags),
+    )
+    argfile = ctx.new_file(ctx.outputs.srcjar, "%s_worker_input" % ctx.label.name)
+    ctx.file_action(output=argfile, content=worker_content)
+    ctx.action(
+        executable = ctx.executable._pluck_scalapb_scala,
+        inputs = list(acc_imports) + [argfile],
+        outputs = [ctx.outputs.srcjar],
+        mnemonic="ProtoScalaRule",
```

Comment:
should this be `ProtoScalaPbRule` or something?

Reply:
Yeah let me rename it